### PR TITLE
2410 add post

### PR DESCRIPTION
**Pull request by rowasc**
This pull request makes the following changes:
- Fix category add feature in add-post
- Fix selection of categories in add-post
- Apply changes for post-editor and data-editor

Testing checklist:
- [ ] Follow test script in #2410

- [ ] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by rowasc** [comment]
screen shot 2018-02-12 at 5 00 15 pm
@Angamanga I noticed there's a style issue in this branch but I can't figure out where it's coming from (it doesn't happen in QA , or in the add post screen, only in the modal where you add categories to a survey) 

Any ideas? :(

File: app/common/directives/category-selector.html [hunk 1]
```diff
@@ -12,13 +12,22 @@
...
                 checklist-change="changeCategories()"
             >
+            <input
+                    ng-show="category.children.length > 0"
+                    type="checkbox"
+                    checklist-value="category.id"
+                    ng-disabled="true"
+                    value="category.id"
+                    checklist-model="selectedParents"
+            >
```
Comment (Angamanga) [inline app/common/directives/category-selector.html]: It looks like the categories doesn't have access to the whole child-objects when I am in the survey-field-category-selector. Not sure why that happened. It doesn't happen in develop :/

File: app/main/posts/modify/post-category-editor.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            form: '='
+        },
+        controller: PostCategoryEditorController,
+        template: require('./post-category-editor.html')
+    };
+}
+PostCategoryEditorController.$inject = ['$rootScope','$scope', 'TagEndpoint', 'FormAttributeEndpoint', 'Notify', '_'];
+
+function PostCategoryEditorController($rootScope, $scope, TagEndpoint, FormAttributeEndpoint, Notify, _) {
+    console.log($scope.available);
```
Comment (Angamanga) [inline app/main/posts/modify/post-category-editor.js]: Remove log

File: app/common/directives/category-selector.html (hunk 1 above)
Comment (rowasc) [reply app/common/directives/category-selector.html]: fixed

Comment (rowasc) [reply app/common/directives/category-selector.html]: > I also didnt exactly understand step nb 6 in the test-checklist: Select only the newly created child category as a filter so maybe I have missed testing something?
this is in another ticket now, we don't have a way to add children in add post screen

**Comment by rowasc** [comment]
https://github.com/ushahidi/platform/issues/2410#issuecomment-365344379
@Angamanga see this

**Pull request merged**